src/Integration.UnitTests/Binding/DotNetBindingConfigProviderTests.cs

```diff
@@ -42,245 +38,302 @@ namespace SonarLint.VisualStudio.Integration.UnitTests
-                NonLocalizedDisplayName = string.Format(Strings.SonarQubeRuleSetNameFormat, ProjectName, QualityProfileName),
-                NonLocalizedDescription = "\r\nhttp://connected/profiles/show?key="
+                Description = OriginalValidRuleSetDescription,
+                ToolsVersion = "12.0",
+                Rules = new List<Core.CSharpVB.Rules>
+                {
+                    new Core.CSharpVB.Rules
+                    {
+                        AnalyzerId = "my analyzer", RuleNamespace = "my namespace"
+                    }
+                }
             };
-            var nugetPackages = new[] { new PackageName("myPackageId", new SemanticVersion("1.0.0")) };
-            var additionalFiles = new[] { new AdditionalFileResponse { FileName = "abc.xml", Content = new byte[] { 1, 2, 3 } } };
-            RoslynExportProfileResponse export = RoslynExportProfileHelper.CreateExport(ruleSet, nugetPackages, additionalFiles);
 
-            var language = Language.VBNET;
-            SonarQubeQualityProfile profile = this.ConfigureProfileExport(export, language, QualityProfileName);
+        [TestMethod]
+        public void GetRules_UnsupportedLanguage_Throws()
+        {
+            var builder = new TestEnvironmentBuilder(ValidQualityProfile, Language.Cpp);
+            var testSubject = builder.CreateTestSubject();
 
             // Act
-            var result = await testSubject.GetConfigurationAsync(profile, "TODO", language, CancellationToken.None)
-                .ConfigureAwait(false);
+            Action act = () => testSubject.GetConfigurationAsync(ValidQualityProfile, null, Language.Cpp, CancellationToken.None).Wait();
 
             // Assert
-            result.Should().NotBeNull();
-            var dotNetResult = result as DotNetBindingConfigFile;
-            dotNetResult.Should().NotBeNull();
-
-            RuleSetAssert.AreEqual(expectedRuleSet, dotNetResult.RuleSet, "Unexpected rule set");
-            VerifyNuGetPackgesDownloaded(nugetPackages, language, actualProfiles);
-
-            this.logger.AssertOutputStrings(0); // not expecting anything in the case of success
+            act.Should().ThrowExactly<ArgumentOutOfRangeException>().And.ParamName.Should().Be("language");
         }
 
         [TestMethod]
-        public async Task GetConfig_WhenProfileExportIsNotAvailable_Fails()
+        public void IsSupported()
         {
             // Arrange
-            var testSubject = this.CreateTestSubject();
-
-            var language = Language.CSharp;
-            var profile = this.ConfigureProfileExport(null, language, "");
+            var builder = new TestEnvironmentBuilder(ValidQualityProfile, Language.Cpp);
+            var testSubject = builder.CreateTestSubject();
 
-            // Act
-            var result = await testSubject.GetConfigurationAsync(profile, null, language, CancellationToken.None)
-                .ConfigureAwait(false);
+            // 1. Supported languages
```

Comment:
You can use DataTestMethod(language, shouldBeSupported).
It's a minor change but it makes the tests a lot more readable, and easier to understand when they fail. It's more descriptive than using comments.

Reply:
Unfortunately `Language` is a class so instances can't be passed in attributes.

Reply:
FYI for review comments like this it's enough write "Why not use DataRow?" - no need to include an explanation.